stats/Record.md

```diff
@@ -69,7 +69,15 @@ private static final MeasureLong RPC_BYTES_SENT =
 MeasurementMap measurementMap = new MeasurementMap();
 measurementMap.put(RPC_LATENCY, 10.3);
 measurementMap.put(RPC_BYTES_SENT, 124);
-measurementMap.withSpanContext(spanContext);
 measurementMap.record();  // reads context from thread-local.
+
+// Another example on recording against sampled SpanContext.
+SpanContext spanContext = tracer.getCurrentSpan().getContext();
+if (spanContext.getTraceOptions().isSampled()) {
+  Map<String, String> attachments = new HashMap<>();
+  // Client code needs to take care of encoding.
+  attachments.put("TraceId", encode(spanContext.getTraceId));
+  attachments.put("SpanId", encode(spanContext.getSpanId));
+  measurementMap.record(attachments);
```

Comment:
I'm not sure if this makes sense, but another way to handle explicit propagation is to pass the whole gRPC context to `record`.  Then the two `record` methods would behave more similarly.  They would both allow the OpenCensus implementation to control how the exemplars are added, rather than pushing that logic into the client code.  I'm not sure how that would work in all languages, though, since opencensus-specs doesn't require a context, like the gRPC context or Go context, that references the SpanContext and TagContext and can be passed around.

Reply:
There could be other exemplar attachments that are not from the current context, like [plain string literals](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries#exemplar). I think our goal here is to make the record() method as generic as possible, and restricting it to take only the current context seems to be against that goal.

Reply:
Then it seems like we should try to make the implicit propagation version of `record` more generic, too, though I'm not sure how that would work.  Maybe both `record` methods should take a `Map<String, String>`, and the implicit propagation version could merge the input map with the keys-value pairs representing the current span.